Client/Frontend/Home/BookmarksPanel.swift

```diff
@@ -195,7 +193,12 @@ class BookmarksPanel: SiteTableViewController, HomePanel {
             if let url = bookmark.favicon?.url.asURL where url.scheme == "asset" {
                 cell.imageView?.image = UIImage(named: url.host!)
             } else {
-                cell.imageView?.setIcon(bookmark.favicon, withPlaceholder: self.defaultIcon)
+                if let bookmarkURL = NSURL(string: item.url) {
+                    cell.imageView?.setIcon(bookmark.favicon, withPlaceholder: FaviconFetcher.getDefaultFavicon(bookmarkURL, f: faviconFetcher))
+                } else {
+                    // Send an empty URL to getDefaulFavicon in order to return the globe icon (since there is no url available)
+                    cell.imageView?.setIcon(bookmark.favicon, withPlaceholder: FaviconFetcher.getDefaultFavicon(NSURL(), f: faviconFetcher))
```

Comment:
This seems too indirect - why not just return back the globe icon instead of going through `FaviconFetcher`?